### PR TITLE
update DSS calls for ClimaCore v0.10.25

### LINKED ISSUE #242: Update DSS usage for ClimaCore v0.10.25
As of ClimaCore v0.10.25, `Spaces.dss_interior_faces!` and `Spaces.dss_local_vertices!` have been replaced with `dss2!`. Both ClimaCoupler and ClimaLSM need to be updated to use the new interface. Specifically, the deprecated functions are used in the `Regridder` code of both packages (which should also be moved to a shared location at some point).

Addressed by #244.

ClimaLSM issue [#144](https://github.com/CliMA/ClimaLSM.jl/issues/144) is the corresponding issue for ClimaLSM.

[Coupler regridder using old functions](https://github.com/CliMA/ClimaCoupler.jl/blob/main/src/Regridder.jl#L50)
[ClimaLSM regridder using old functions](https://github.com/CliMA/ClimaLSM.jl/blob/main/src/Regridder.jl#L50)
[new dss2! function](https://github.com/CliMA/ClimaCore.jl/blob/main/src/Spaces/dss2.jl#L998)
[example using new dss2! which was previously using old functions](https://github.com/CliMA/ClimaCore.jl/blob/95c84fcaf3dc557ff1ebd7c76fc89e2b413efdba/lib/ClimaCoreTempestRemap/src/onlineremap.jl#L59)
[Coupler deleted commit with replaced functions](https://github.com/CliMA/ClimaCoupler.jl/commit/bdb4dc6d3963853149c0e97bdb4250ab1dba36bb#diff-1580d3871984bc0075f5e020050fe2c92cdabfa720c6c16ae20146f7f9ebbb2a)

NB:
- [ ] let Sriharsha know once this is done.

### PR DESCRIPTION
**Pull request by juliasloan25**
## Purpose 
Replace function calls to `Spaces.dss_interior_faces!` and `Spaces.dss_local_vertices!` with `Spaces.dss2!` to use the new ClimaCore interface from v0.10.25.

Note that other dss function calls may remain the same thanks to backwards compatibility implemented [here](https://github.com/CliMA/ClimaCore.jl/blob/main/src/Fields/Fields.jl#L347).

Closes #242.

## To-do
- [x] Replace calls to `dss_interior_faces!` and `dss_local_vertices!` with `dss2!`
- [x] Update Project.toml ClimaCore compat entries

----
- [x] I have read and checked the items on the review checklist.

### CODE REVIEW

**Comment by juliasloan25** [comment]
bors try

**Comment by juliasloan25** [comment]
bors try

**Comment by simonbyrne** [comment]
The issues is here:
https://buildkite.com/clima/climacoupler-ci/builds/923#01863821-062c-4fd8-8148-41f52cdcb01f/140-287

you're passing a `ReshapedArray` to the 3rd arg of this struct:
https://github.com/CliMA/ClimaCoupler.jl/blob/82cb858987da98cb343e6f904a9e494772be953e/src/PostProcessor.jl#L30-L35

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -8,6 +8,7 @@ env:
   BUILDKITE_COMMIT: "${BUILDKITE_COMMIT}"
   BUILDKITE_BRANCH: "${BUILDKITE_BRANCH}"
   JULIA_MAX_NUM_PRECOMPILE_FILES: 100
+  GKSwstype: 100
```
Comment (LenkaNovak) [inline .buildkite/pipeline.yml:11]: Nice, I see this fixes the GKS warnings! 👍

**Review by LenkaNovak — APPROVED** [review]
Looks great, Julia!

**Pull request merged**